src/ProgressOnderwijsUtils/SingleSignOn/SsoProcessor.cs

```diff
@@ -114,7 +114,7 @@ static NameValueCollection CreateQueryString(AuthnRequest req, [CanBeNull] strin
         static string Signature([NotNull] NameValueCollection qs, AsymmetricAlgorithm key)
         {
             var data = Encoding.UTF8.GetBytes(ToQueryString(qs));
-            var result = ((RSACryptoServiceProvider)key).SignData(data, new SHA1CryptoServiceProvider());
+            var result = ((RSA)key).SignData(data, HashAlgorithmName.SHA1, RSASignaturePadding.Pkcs1);

```

Comment:
waarom is hier ueberhaupt een cast nodig?  waarom is key slecht typeerd?

Reply:
omdat een pfx blijkbaar ook andere dingen kan bevatten

Reply:
Ik zou de cast bij de caller verwachten, niet hier, omdat je dus geen enkel ander type kan ondersteunen, en het weinig zin heeft zo laat pas mis te gaan bij type-mismatch (en dan kan geen caller het idee krijgen dan een andere AsymmetricAlgorithm kan werken).

Reply:
Klinkt redelijk, maar omdat het een private method is is mijn medeleven beperkt.